Modules/_abc.c

```diff
@@ -65,24 +80,30 @@ abc_data_new(PyTypeObject *type, PyObject *args, PyObject *kwds)
 PyDoc_STRVAR(abc_data_doc,
 "Internal state held by ABC machinery.");
 
-static PyTypeObject _abc_data_type = {
-    PyVarObject_HEAD_INIT(NULL, 0)
-    "_abc_data",                        /*tp_name*/
-    sizeof(_abc_data),                  /*tp_basicsize*/
-    .tp_dealloc = (destructor)abc_data_dealloc,
-    .tp_flags = Py_TPFLAGS_DEFAULT,
-    .tp_alloc = PyType_GenericAlloc,
-    .tp_new = abc_data_new,
+static PyType_Slot _abc_data_type_spec_slots[] = {
+    {Py_tp_doc, (void *)abc_data_doc},
+    {Py_tp_dealloc, abc_data_dealloc},
+    {Py_tp_alloc, PyType_GenericAlloc},
+    {Py_tp_new, abc_data_new},
```

Comment:
PyType_GenericAlloc is the default, there is no need to override tp_alloc. While we modify the slots, I prefer to see alloc before dealloc :-)

```suggestion
    {Py_tp_doc, (void *)abc_data_doc},
    {Py_tp_new, abc_data_new},
    {Py_tp_dealloc, abc_data_dealloc},

```